### PR TITLE
Add benchmark script

### PR DESCRIPTION
**Pull request by AlexWayfer**
Can be executed by `bundle exec ruby benchmark.rb` or `bundle exec rake benchmark`.

### CODE REVIEW

File: benchmark.rb [hunk 1]
```diff
@@ -0,0 +1,46 @@
+# frozen_string_literal: true
+
+require "bundler/setup"
+Bundler.setup
+
+require "benchmark"
+require "benchmark/ips"
+require "benchmark/memory"
+
+puts "```ruby"
```
Comment (tycooon) [inline benchmark.rb]: What's the purpose of this `puts` stuff?

Comment (AlexWayfer) [reply benchmark.rb]: You just copy the full output, paste it in GitHub, for example — that's all.

Example:

> bundle exec rake benchmark
```ruby
# frozen_string_literal: true

require "bundler/setup"
Bundler.setup

require "benchmark"
require "benchmark/ips"
require "benchmark/memory"

puts "```ruby"
puts File.read(__FILE__)
puts "```"
puts
puts "### Output"
puts
puts "```"

require_relative "lib/memery"

class Foo
  class << self
    include Memery

    def base_find(char)
      ("a".."k").find { |letter| letter == char }
    end

    memoize def find_new(char)
      base_find(char)
    end
  end
end

def test_memery
  Foo.find_new("d")
end

Benchmark.ips do |x|
  x.report("test_memery") { test_memery }
end

Benchmark.memory do |x|
  x.report("test_memery") { 100.times { test_memery } }
end

puts "```"
```

### Output

```
Warming up --------------------------------------
         test_memery    33.193k i/100ms
Calculating -------------------------------------
         test_memery    386.577k (± 2.4%) i/s -      1.958M in   5.069159s
Calculating -------------------------------------
         test_memery    16.000k memsize (     0.000  retained)
                       400.000  objects (     0.000  retained)
                         3.000  strings (     0.000  retained)
```

Comment (AlexWayfer) [reply benchmark.rb]: The bench script itself and bench results in one place, Markdown-formatted.

Comment (tycooon) [reply benchmark.rb]: 👍

**Pull request merged**